modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceRegexCallbackUrlTestCase.java

```diff
@@ -0,0 +1,213 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.identity.integration.test.oauth2;
+
+import org.apache.catalina.startup.Tomcat;
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+import org.apache.http.NameValuePair;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.identity.oauth.stub.dto.OAuthConsumerAppDTO;
+import org.wso2.carbon.integration.common.admin.client.AuthenticatorClient;
+import org.wso2.identity.integration.test.utils.DataExtractUtil;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import java.io.File;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class OAuth2ServiceRegexCallbackUrlTestCase extends OAuth2ServiceAbstractIntegrationTest {
+
+	private AuthenticatorClient logManger;
+	private String adminUsername;
+	private String adminPassword;
+	private String accessToken;
+	private String sessionDataKeyConsent;
+	private String sessionDataKey;
+
+	private String consumerKey;
+	private String consumerSecret;
+
+	private DefaultHttpClient client;
+	private Tomcat tomcat;
+
+	@BeforeClass(alwaysRun = true)
+	public void testInit() throws Exception {
+		super.init(TestUserMode.SUPER_TENANT_USER);
+		logManger = new AuthenticatorClient(backendURL);
+		adminUsername = userInfo.getUserName();
+		adminPassword = userInfo.getPassword();
+		logManger.login(isServer.getSuperTenant().getTenantAdmin().getUserName(),
+				isServer.getSuperTenant().getTenantAdmin().getPassword(),
+				isServer.getInstance().getHosts().get("default"));
+
+		setSystemproperties();
+		client = new DefaultHttpClient();
+	}
+
+	@AfterClass(alwaysRun = true)
+	public void atEnd() throws Exception {
+		deleteApplication();
+		removeOAuthApplicationData();
+		stopTomcat(tomcat);
+
+		logManger = null;
+		consumerKey = null;
+		accessToken = null;
+	}
+
+	@Test(alwaysRun = true, description = "Deploy playground application")
+	public void testDeployPlaygroundApp() {
+		try {
+			tomcat = getTomcat();
+			URL resourceUrl =
+					getClass().getResource(File.separator + "samples" + File.separator +
+							"playground2.war");
+			startTomcat(tomcat, OAuth2Constant.PLAYGROUND_APP_CONTEXT_ROOT, resourceUrl.getPath());
+
+		} catch (Exception e) {
+			Assert.fail("Playground application deployment failed.", e);
+		}
+	}
+
+	@Test(groups = "wso2.is", description = "Check Oauth2 application registration", dependsOnMethods = "testDeployPlaygroundApp")
+	public void testRegisterApplication() throws Exception {
```

Comment:
This is somewhat different application where callback URL contains a regular expression rather than a normal URL. Hence IMO it's better to have this.